Makefile

```diff
@@ -1,5 +1,67 @@
-build: webui
-	go build
+
+EXECUTABLE := qitmeer-wallet
+GITVER := $(shell git rev-parse --short=7 HEAD )
+GITDIRTY := $(shell git diff --quiet || echo '-dirty')
+GITVERSION = "$(GITVER)$(GITDIRTY)"
+DEV=dev
+RELEASE=release
+LDFLAG_DEV = -X github.com/Qitmeer/qitmeer-wallet/version.AppBuild=$(DEV)-$(GITVERSION)
+LDFLAG_RELEASE = -X github.com/Qitmeer/qitmeer-wallet/version.AppBuild=$(RELEASE)-$(GITVERSION)
+GOFLAGS_DEV = -ldflags "$(LDFLAG_DEV)"
+GOFLAGS_RELEASE = -ldflags "$(LDFLAG_RELEASE)"
+VERSION=$(shell ./build/bin/qitmeer-wallet --version | grep ^qitmeer-wallet | cut -d' ' -f3|cut -d'+' -f1)
+GOBIN = ./build/bin
+
+UNIX_EXECUTABLES := \
+	build/release/darwin/amd64/bin/$(EXECUTABLE) \
+	build/release/linux/amd64/bin/$(EXECUTABLE)
+WIN_EXECUTABLES := \
+	build/release/windows/amd64/bin/$(EXECUTABLE).exe
+
+EXECUTABLES=$(UNIX_EXECUTABLES) $(WIN_EXECUTABLES)
+
+COMPRESSED_EXECUTABLES=$(UNIX_EXECUTABLES:%=%.tar.gz) $(WIN_EXECUTABLES:%.exe=%.zip) $(WIN_EXECUTABLES:%.exe=%.cn.zip)
+
+RELEASE_TARGETS=$(EXECUTABLES) $(COMPRESSED_EXECUTABLES)
+
```

Comment:
.PHONY: build

Reply:
I'm sorry, I didn't understand what you meant, could you elaborate

Reply:
A phony target is one that is not really the name of a file,  see https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html for more details.
The `.PHONY` make sure ‘make build’ will run the recipe regardless of whether there is a folder `build` is touched.

Reply:
ok